packages/loopring_v3/test/testUserStakingPool.ts

```diff
@@ -1,266 +1,307 @@
+    const amount = new BN("1000" + "000000000000000000", 10);
+
+    it("can stake 1000 LRC then query getUserStaking", async () => {
+      const tx = await userStakingPool.stake(amount, { from: bob });
+
+      // - Check: LRCStaked emitted
+      truffleAssert.eventEmitted(tx, "LRCStaked", (evt: any) => {
+        return bob === evt.user && amount.eq(evt.amount);
+      });
+
+      // - Check: stats
+      const {
+        0: withdrawalWaitTime,
+        1: rewardWaitTime,
+        2: balance,
+        3: claimableReward
+      } = await userStakingPool.getUserStaking(bob);
+
+      assert(
+        withdrawalWaitTime.eq(new BN(MIN_WITHDRAW_DELAY)),
+        "withdrawalWaitTime"
       );
+      assert(rewardWaitTime.eq(new BN(MIN_CLAIM_DELAY)), "rewardWaitTime");
+      assert(balance.eq(amount), "balance");
+      assert(claimableReward.eq(ZERO), "claimableReward");
     });
-    it("should get user staking equal to staked", async () => {
-      await userstaking.stake(1000, { from: owner1 });
-      const userstaked = await userstaking.getUserStaking(owner1);
-      assert.equal(
-        userstaked.balance,
-        1000,
-        "User staking should equal to expected"
+
+    it("can query getUserStaking after MIN_WITHDRAW_DELAY/2 seconds", async () => {
+      await advanceTimeAndBlockAsync(MIN_WITHDRAW_DELAY / 2);
+
+      // - Check: stats
+      const {
+        0: withdrawalWaitTime,
+        1: rewardWaitTime,
+        2: balance,
+        3: claimableReward
+      } = await userStakingPool.getUserStaking(bob);
+
+      assert(
+        withdrawalWaitTime.eq(new BN(MIN_WITHDRAW_DELAY / 2)),
+        "withdrawalWaitTime"
       );
+      assert(
+        rewardWaitTime.eq(new BN(MIN_CLAIM_DELAY - MIN_WITHDRAW_DELAY / 2)),
+        "rewardWaitTime"
+      );
+      assert(balance.eq(amount), "balance");
+      assert(claimableReward.eq(ZERO), "claimableReward");
     });
-    it("should get total staking equal to staked", async () => {
-      const userstaked = await userstaking.getTotalStaking();
-      assert.equal(userstaked, 1000, "Total staking should equal to expected");
-    });
-    it("should NEED_TO_WAIT when withdraw", async () => {
-      await expectThrow(
-        userstaking.withdraw(500, { from: owner1 }),
-        "NEED_TO_WAIT"
+
+    it("can query getUserStaking 1 second before timeout but still cannot withdraw", async () => {
+      await advanceTimeAndBlockAsync(MIN_WITHDRAW_DELAY / 2 - 1);
+
+      // - Check: stats
+      const {
+        0: withdrawalWaitTime,
+        1: rewardWaitTime,
+        2: balance,
+        3: claimableReward
+      } = await userStakingPool.getUserStaking(bob);
+
+      assert(withdrawalWaitTime.eq(new BN(1)), "withdrawalWaitTime");
+      assert(rewardWaitTime.eq(new BN(1)), "rewardWaitTime");
+      assert(balance.eq(amount), "balance");
+      assert(claimableReward.eq(ZERO), "claimableReward");
+
+      // - Check: withdrawal will fail
+      await truffleAssert.fails(
```

Comment:
This can check against the actual revert message using `expectThrow` or `truffleAssert.reverts`. This way it can be sure the function reverts for the correct reason.

Reply:
Will also do it in my next PR which focus on this test file.